internal/dependency/default_repo_adapter.go

```diff
@@ -13,41 +15,39 @@ type defaultRepoAdapter struct {
 }
 
 func (a *defaultRepoAdapter) GetVersions(name string) ([]string, error) {
-	packageRepo, err := a.getRepoForPackage(name)
-	if err != nil {
-		return nil, err
+	packageRepo, repoErr := a.getRepoForPackage(name)
+	if repoErr != nil && packageRepo == nil {
+		return nil, repoErr
 	}
 	var idx repotypes.PackageIndex
 	if err := a.client.ForRepo(*packageRepo).FetchPackageIndex(name, &idx); err != nil {
-		return nil, err
+		return nil, multierr.Append(err, repoErr)
```

Comment:
Nit: I think it's common practice to order the errors from oldest to newest
```suggestion
		return nil, multierr.Append(repoErr, err)
```